feature/preference/src/main/java/io/github/droidkaigi/confsched2020/preference/ui/PreferencesFragment.kt

```diff
@@ -52,6 +46,40 @@ class PreferencesFragment : PreferenceFragmentCompat() {
         val component = DaggerPreferenceComponent.factory()
             .create(appComponent, PreferenceModule(this))
         component.inject(this)
+
+        preferenceManager?.findPreference<SwitchPreferenceCompat>(SWITCH_DARK_THEME_KEY)?.also {
+            val currentNightMode = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
+            it.isChecked = currentNightMode == Configuration.UI_MODE_NIGHT_YES
+            preferenceViewModel.setNightMode(it.isChecked)
+            it.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, newValue ->
+                if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.P) {
```

Comment:
Thanks!
I do not understand well, is it not possible to observe and update onChanged of UiModel on both OSs?

Reply:
I want to make sure that my understanding is that the dark theme setting of the app meant to be the same as the device setting (for Android 10 and above)

Therefore I am having issue when the user switch on the dark theme when the app is running on the preference fragment, the theme wouldn’t reflect the configuration in previous implementation.

There might be a workaround by changing the order of the code but I don’t have time to test it out yet...

Will work on it ASAP 
Sorry for having so many concern since this is the first time I contribute to OSS😢

Reply:
The issue I mentioned is shown in the following GIF, this only happens in Android 10 since it allows user to change the system theme setting
![Uploading android10.gif…](issue)